one-pager-maker/src/pages/Edit.tsx

```diff
@@ -33,9 +33,10 @@ function Edit() {
   const reviewHistoryMutation = viewHistoryApi.useSetReviewHistoryMutation();
 
   useEffect(() => {
```

Comment:
This might be good. I want to try it locally.
```
  useEffect(() => {
    if (document) {
      setDocumentData(document);
    }
  }, []);
```

Reply:
@binnmti 
Thank you for review!!

I agree that your code can preserve data when switching tabs.
However, when it reloads, data will not be preserved.
That's why I set documentData on useEffect method.

Here's the reproduction:

https://github.com/user-attachments/assets/0eddf9a2-0321-4a29-a0b1-15810c1e86b0

Reply:
this is?
```
  useEffect(() => {
    if (document) {
      setDocumentData(document);
    }
  }, [document]);
```

Reply:
That code doesn't work as intended, because it resets user input every time the `document` changes.
useEffect runs on every `document` update, overwriting documentData with the original document state.

Reply:
I changed the code like this:

``` ts
useEffect(() => {
    if (document && !documentData) {
      setDocumentData(document);
    }
  }, [document, documentData]);
```

It'll be more easy to read I think

Reply:
ありがとうございます。数回試した場合はセーフでしたが、再度試したら確かに消えました。。

すいません。もう1回だけ。赤いの消しても動かないですか？
![image](https://github.com/user-attachments/assets/dcf81441-8431-4cfb-82c2-bf7e1166ee1e)

Reply:
@binnmti 
タブ移動やリロードに対応できると思います！

あえて残している理由としては依存配列には使用される全てのものを入れるべきだと書いてあったからです！
https://ja.react.dev/reference/react/useEffect#specifying-reactive-dependencies

Reply:
色々間違った指摘をしてゴメンナサイ。対応してくれてありがとうございます！

Reply:
僕もレビューから初めて気づくところが多かったので、全然間違ったと思っていません!!
ありがとうございました🙇‍♂️🙇‍♂️